cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/aws/aws-mc-common/aws-orchestrator/AwsOrchestrator.service.tsx

```diff
@@ -0,0 +1,48 @@
+// React imports
+import { AwsService, AWSVirtualMachine } from '../../../../../swagger-api';
+import { AWS_FIELDS } from '../../aws-mc-basic/AwsManagementClusterBasic.constants';
+import { INPUT_CHANGE } from '../../../../../state-management/actions/Form.actions';
+import { AwsDefaults } from '../default-service/AwsDefaults.service';
+import { STORE_SECTION_FORM } from '../../../../../state-management/reducers/Form.reducer';
+import { AWS_ADD_RESOURCES } from '../../../../../state-management/actions/Resources.actions';
+import { AwsResourceAction, FormAction, StoreDispatch } from '../../../../../shared/types/types';
+interface AwsOrchestratorProps {
+    awsState: { [key: string]: any };
+    awsDispatch: StoreDispatch;
+}
+export class AwsOrchestrator {
+    static async initOsImages(props: AwsOrchestratorProps) {
+        // clear previous os images
+        clearOsImages(props);
+        // retrieve os images
+        const osImages = await AwsService.getAwsosImages(props.awsState[STORE_SECTION_FORM].REGION);
```

Comment:
For the next PR... how do we want to handle errors?

Reply:
Yes sir. So in _ManangementCredentials.tsx_ , I use `try catch` in line 162 ( this can be found in this PR)

Reply:
I'm a little uncertain here without testing the implementation, but since this sits in an async function and we are catching the error in ManagementCredentials.tsx, I _think_ that would catch the api error. Are you thinking we should do some additional handling here in the orchestrator?

Reply:
Whoops you beat me to it He. Your comment appeared after I left mine